### PR TITLE
Alfredapi-346

### PR DESCRIPTION
**Pull request by WimCrols**
Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-346

- [x] Is [CHANGELOG.md](./CHANGELOG.md) extended?
- [x] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [x] Does the PR comply to REST HTTP result codes policy outlined in the [developer guide](./developer-documentation)?
- [x] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [x] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-
 
-- [ ] Is [CHANGELOG.md](./CHANGELOG.md) extended?
+- [ ] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
 - [ ] Does this PR avoid breaking the API? 
     Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
-- [ ] Does the PR comply to REST HTTP result codes policy outlined in the [developer guide](./developer-documentation)?
+- [ ] Does the PR comply to REST HTTP result codes policy outlined in the [developer guide](https://github.com/xenit-eu/alfred-api/blob/master/developer-documentation)?
 - [ ] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
```
Comment (rschev) [inline .github/pull_request_template.md]: We should probably publicize the code style guide somewhere external people can view it too.

File: Jenkinsfile [hunk 2]
```diff
@@ -26,18 +26,12 @@ def sendEmailNotifications() {
...
-        return 'Snapshot'
-    }
-    return null
+def isPublishNeeded() {
+    def gitBranch = env.BRANCH_NAME
+    return gitBranch.startsWith("release") || gitBranch == "master"
 }
 
-def BuildVersionX(publishingRepo, version) {
+def BuildVersionX(isPublishNeeded, version) {
```
Comment (rschev) [inline Jenkinsfile]: I would avoid shadowing the function with a boolean, I got pretty confused thinking why you would want to pass around a function here.

File: Jenkinsfile [hunk 3]
```diff
@@ -66,15 +59,15 @@ def BuildVersionX(publishingRepo, version) {
 
 node {
     def gradleCommand = "./gradlew --info --stacktrace "
-    def publishingRepo = getPublishingRepository()
+    def isPublishNeeded = isPublishNeeded()
```
Comment (rschev) [inline Jenkinsfile]: Is there still any functionality that makes a distinction between publishing snapshots and release artifacts? Does Sonatype just detect it correctly?

File: .github/pull_request_template.md (hunk 1 above)
Comment (WimCrols) [reply .github/pull_request_template.md]: Yes, if only out of principle (no-one will ever commit on apix), but out of scope for this ticket.

File: Jenkinsfile (hunk 2 above)
Comment (WimCrols) [reply Jenkinsfile]: Ok, will adjust.

File: Jenkinsfile (hunk 3 above)
Comment (WimCrols) [reply Jenkinsfile]: The logic is moved to gradle (That's something in general we want to do)

**Pull request merged**